site/src/components/Spinner/Spinner.tsx

```diff
@@ -1,22 +1,77 @@
-import CircularProgress, {
-	type CircularProgressProps,
-} from "@mui/material/CircularProgress";
-import isChromatic from "chromatic/isChromatic";
-import type { FC } from "react";
-
 /**
- * Spinner component used to indicate loading states. This component abstracts
- * the MUI CircularProgress to provide better control over its rendering,
- * especially in snapshot tests with Chromatic.
+ * This component was inspired by
+ * https://www.radix-ui.com/themes/docs/components/spinner and developed using
+ * https://v0.dev/ help.
  */
-export const Spinner: FC<CircularProgressProps> = (props) => {
-	/**
-	 * During Chromatic snapshots, we render the spinner as determinate to make it
-	 * static without animations, using a deterministic value (75%).
-	 */
-	if (isChromatic()) {
-		props.variant = "determinate";
-		props.value = 75;
+
+import { type VariantProps, cva } from "class-variance-authority";
+import type { ReactNode } from "react";
+import { cn } from "utils/cn";
+import isChromatic from "chromatic/isChromatic";
+
+const leaves = 8;
+
+const spinnerVariants = cva("", {
+	variants: {
+		size: {
+			lg: "size-icon-lg",
+			sm: "size-icon-sm",
+		},
+	},
+	defaultVariants: {
+		size: "lg",
+	},
+});
+
+type SpinnerProps = React.SVGProps<SVGSVGElement> &
+	VariantProps<typeof spinnerVariants> & {
+		children?: ReactNode;
+		loading?: boolean;
+	};
+
+export function Spinner({
+	className,
+	size,
+	loading,
+	children,
+	...props
+}: SpinnerProps) {
+	if (!loading) {
+		return children;
 	}
-	return <CircularProgress {...props} />;
-};
+
+	return (
+		<svg
+			viewBox="0 0 24 24"
+			xmlns="http://www.w3.org/2000/svg"
+			fill="currentColor"
+			className={cn(spinnerVariants({ size, className }))}
+			{...props}
+		>
+			<title>Loading spinner</title>
+			{[...Array(leaves)].map((_, i) => {
+				const rotation = i * (360 / leaves);
+
+				return (
+					<rect
+						key={i}
+						x="11.5"
+						y="3"
+						width="1.5"
+						height="4"
+						rx="0.5"
```

Comment:
```suggestion
						x="10.9"
						y="2"
						width="2"
						height="5.5"
						rx="1"
```

This change will make it closer to the Figma design